docs/content/guides/dagster-pipes/subprocess/reference.mdx

```diff
@@ -256,3 +256,76 @@ defs = Definitions(
 
 </TabItem>
 </TabGroup>
+
+---
+
+## Passing custom data
+
+Sometimes, you may want to pass data back from the external process for use in the orchestration code for purposes other than reporting directly to Dagster.
+
+<TabGroup>
+<TabItem name="External code in external_code.py">
+
+In the external code, we send messages using `report_custom_message`. The message can be any data that is JSON serializable.
+
+```python file=/guides/dagster/dagster_pipes/subprocess/custom_messages/external_code.py
+from dagster_pipes import PipesContext, open_dagster_pipes
+
+
+def main():
+    # get the Dagster Pipes context
+    context = PipesContext.get()
+    data = {"type": "example_msg", "data": 42}  # send any JSON serializable data
+    context.report_custom_message(data)
+
+
+if __name__ == "__main__":
+    # connect to Dagster Pipes
+    with open_dagster_pipes():
+        main()
+```
+
+</TabItem>
+<TabItem name="Dagster code in dagster_code.py">
+
+In the Dagster code we receive custom messages using `get_custom_messages`.
+
+```python file=/guides/dagster/dagster_pipes/subprocess/custom_messages/dagster_code.py
+import shutil
+
+from dagster import (
+    Definitions,
+    OpExecutionContext,
+    PipesSubprocessClient,
+    file_relative_path,
+    job,
+    op,
+)
+
+
+@op
+def subprocess_op(context: OpExecutionContext, pipes_client: PipesSubprocessClient):
```

Comment:
Why ops over assets here?

Reply:
the thinking was it would be good to diversify our examples and show that this stuff can be used in ops as well but I can certainly change it

Reply:
Perhaps we can change the example to assets since that's the focus, and mention that this is also possible with ops?